### ISSUE TITLE
Android app crashing when opening second window

### ISSUE THREAD

**Issue by andydotxyz** [fyne-io/fyne#4072]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

Create an app with two windows, open the second one on an android device and the app will crash.

### How to reproduce

run code below, tap "Button" observe crash:

```
03-02 01:57:14.001 28015     0 E Go      : panic: failed to link OpenGL program; error code: 501
03-02 01:57:14.001 28015     0 E Go      : 
03-02 01:57:14.001 28015     0 E Go      : goroutine 14 [running]:
03-02 01:57:14.001 28015     0 E Go      : fyne.io/fyne/v2/internal/painter/gl.(*painter).createProgram(0x4000184ea0, {0x7c07756b7d?, 0x0?})
03-02 01:57:14.001 28015     0 E Go      :      /Users/andy/Code/Fyne/fyne/internal/painter/gl/painter.go:179 +0x594
03-02 01:57:14.001 28015     0 E Go      : fyne.io/fyne/v2/internal/painter/gl.(*painter).Init(0x4000184ea0)
03-02 01:57:14.001 28015     0 E Go      :      /Users/andy/Code/Fyne/fyne/internal/painter/gl/gl_gomobile.go:71 +0x144
03-02 01:57:14.001 28015     0 E Go      : fyne.io/fyne/v2/internal/driver/mobile.(*mobileDriver).handlePaint(0x4000400000, {0x80?}, {0x7c07d9eaf8, 0x4000326af0})
03-02 01:57:14.001 28015     0 E Go      :      /Users/andy/Code/Fyne/fyne/internal/driver/mobile/driver.go:250 +0x8c
03-02 01:57:14.001 28015     0 E Go      : fyne.io/fyne/v2/internal/driver/mobile.(*mobileDriver).Run.func1({0x7c07d9beb8?, 0x7c0873c560})
03-02 01:57:14.001 28015     0 E Go      :      /Users/andy/Code/Fyne/fyne/internal/driver/mobile/driver.go:188 +0x394
03-02 01:57:14.001 28015     0 E Go      : fyne.io/fyne/v2/internal/driver/mobile/app.mainUI.func1()
03-02 01:57:14.001 28015     0 E Go      :      /Users/andy/Code/Fyne/fyne/internal/driver/mobile/app/android.go:429 +0x48
03-02 01:57:14.001 28015     0 E Go      : created by fyne.io/fyne/v2/internal/driver/mobile/app.mainUI
03-02 01:57:14.001 28015     0 E Go      :      /Users/andy/Code/Fyne/fyne/internal/driver/mobile/app/android.go:428 +0x98
```

### Example code

```go
package main

import (
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/widget"
)

func main() {
	application := app.NewWithID("testui")
	window1 := application.NewWindow("Window One")

	window1.SetContent(widget.NewButton("Button", func() {
		window2 := application.NewWindow("Window Two")
		window2.SetContent(widget.NewLabel("test"))
		window2.Show()
	}))
	window1.Show()
	application.Run()
}
```

### Fyne version

2.3.x, 2.2.x, develop

### Go compiler version

1.20.4

### Operating system and version

macOS M2

**Comment by andydotxyz** [comment]
It would appear that we are re-compiling programs into re-used contexts... the following quick patch seems to revive the functionality. However I think the complete fix needs to be rather more robust! Perhaps we can ask if the gl context has a program already compiled?

```diff
--- a/internal/painter/gl/gl_gomobile.go
+++ b/internal/painter/gl/gl_gomobile.go
@@ -64,14 +64,24 @@ func (p *painter) glctx() gl.Context {
        return p.contextProvider.Context().(gl.Context)
 }
 
+var a, b, c, d Program
+
 func (p *painter) Init() {
        p.ctx = &mobileContext{glContext: p.contextProvider.Context().(gl.Context)}
        p.glctx().Disable(gl.DepthTest)
        p.glctx().Enable(gl.Blend)
-       p.program = p.createProgram("simple_es")
-       p.lineProgram = p.createProgram("line_es")
-       p.rectangleProgram = p.createProgram("rectangle_es")
-       p.roundRectangleProgram = p.createProgram("round_rectangle_es")
+       if !done {
+               done = true
+
+               a = p.createProgram("simple_es")
+               b = p.createProgram("line_es")
+               c = p.createProgram("rectangle_es")
+               d = p.createProgram("round_rectangle_es")
+       }
+       p.program = a
+       p.lineProgram = b
+       p.rectangleProgram = c
+       p.roundRectangleProgram = d
 }
 
```

**Comment by andydotxyz** [comment]
Resolved on develop for v2.4.0
